pkg/kubelet/dockertools/labels.go

```diff
@@ -190,23 +207,90 @@ func getInt64PointerFromLabel(labels map[string]string, label string) (*int64, e
 func getJsonObjectFromLabel(labels map[string]string, label string, value interface{}) (bool, error) {
 	if strValue, found := labels[label]; found {
 		err := json.Unmarshal([]byte(strValue), value)
+		if err != nil {
+			err = parseLabelError(label, strValue, err)
+		}
 		return found, err
 	}
 	// Because it's normal that a container has no PreStopHandler label, don't report any error here.
 	return false, nil
 }
 
-// The label kubernetesPodLabel is added a long time ago (#7421), it serialized the whole api.Pod to a docker label.
+func logLabelError(containerInfo labelledContainerInfo, label string, err error) {
+	glog.Errorf("Unable to get %q for container %q of pod %q: %v", label, containerInfo.Name,
+		kubecontainer.BuildPodFullName(containerInfo.PodName, containerInfo.PodNamespace), err)
+}
+
+func parseLabelError(label string, value string, err error) error {
+	return fmt.Errorf("unable to parse label %q: value=%q, error=%v", label, value, err)
+}
+
+// Backward compatible labels for kubernetes v1.2.
+// TODO(random-liu): These labels don't follow the docker label naming convention, deprecate this
+// when we drop support for v1.2.
+const (
+	oldPodDeletionGracePeriodLabel          = "io.kubernetes.pod.deletionGracePeriod"
+	oldPodTerminationGracePeriodLabel       = "io.kubernetes.pod.terminationGracePeriod"
+	oldContainerRestartCountLabel           = "io.kubernetes.container.restartCount"
+	oldContainerTerminationMessagePathLabel = "io.kubernetes.container.terminationMessagePath"
+	oldContainerPreStopHandlerLabel         = "io.kubernetes.container.preStopHandler"
+)
+
+// Both kubernetes and docker recommend to only use [a-z0-9-.] in label keys, so we decide to replace the old labels.
+// The old labels are added before v1.2, and should be completely removed after we drop support for v1.2.
+// TODO(random-liu): Remove this function when we drop support for v1.2.
+func supplyContainerInfoWithV12Label(labels map[string]string, containerInfo *labelledContainerInfo) {
```

Comment:
IIUC, "V12" is used to identify that the labels are used by kubelet v1.2. What do we gain by changing this to "Deprecated"?